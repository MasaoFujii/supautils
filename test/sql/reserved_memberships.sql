-- use a role that has the CREATEROLE privilege
set role rolecreator;
\echo

-- cannot grant reserved memberships
grant pg_read_server_files to anon;
grant pg_write_server_files to anon;
grant pg_execute_server_program to anon;
\echo

-- can grant non-reserved memberships
grant pg_monitor to anon;
grant pg_read_all_settings to anon;
grant pg_read_all_stats to anon;
