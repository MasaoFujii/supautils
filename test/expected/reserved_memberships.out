-- use a role that has the CREATEROLE privilege
set role rolecreator;
\echo

-- cannot grant reserved memberships
grant pg_read_server_files to anon;
ERROR:  Only superusers can GRANT "pg_read_server_files"
grant pg_write_server_files to anon;
ERROR:  Only superusers can GRANT "pg_write_server_files"
grant pg_execute_server_program to anon;
ERROR:  Only superusers can GRANT "pg_execute_server_program"
\echo

-- can grant non-reserved memberships
grant pg_monitor to anon;
grant pg_read_all_settings to anon;
grant pg_read_all_stats to anon;
